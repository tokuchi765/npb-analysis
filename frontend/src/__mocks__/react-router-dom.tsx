import React from 'react';
const rrd = require('react-router-dom');
// Just render plain div with its children
rrd.BrowserRouter = ({ children }: any) => <div>{children}</div>;
module.exports = rrd;
