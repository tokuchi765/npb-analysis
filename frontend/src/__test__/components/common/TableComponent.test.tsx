import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {
  TableComponent,
  TableLinkComponent,
  HeadCell,
  SelectItem,
} from '../../../components/common/TableComponent';
import { MemoryRouter } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() });

describe('TableComponentテスト', () => {
  it('スナップショット作成', () => {
    const headCells: HeadCell[] = [
      { id: 'main', numeric: false, disablePadding: true, label: 'チーム名' },
      { id: 'battingAverage', numeric: true, disablePadding: false, label: '打率' },
    ];
    interface TestData {
      main: string;
      battingAverage: number;
    }

    const testData: TestData = { main: 'メイン', battingAverage: 0.3 };
    const tree = renderer
      .create(
        <TableComponent
          title={'タイトル'}
          setSelect={() => {}}
          getDataList={() => {}}
          datas={[testData]}
          headCells={headCells}
          initSorted={'初期ソート'}
          selectItems={[new SelectItem('選択1', 'ラベル名', ['選択1', '選択2'])]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TableLinkComponentテスト', () => {
  it('スナップショット作成', () => {
    const headCells: HeadCell[] = [
      { id: 'main', numeric: false, disablePadding: true, label: 'チーム名' },
      { id: 'battingAverage', numeric: true, disablePadding: false, label: '打率' },
    ];
    interface TestData {
      main: string;
      battingAverage: number;
    }

    const testData: TestData = { main: 'メイン', battingAverage: 0.3 };
    const playerIdMap: Map<string, string> = new Map<string, string>();
    playerIdMap.set('key', 'value');
    const tree = renderer
      .create(
        <MemoryRouter>
          <TableLinkComponent
            title={'タイトル'}
            setSelect={() => {}}
            getDataList={() => {}}
            datas={[testData]}
            headCells={headCells}
            initSorted={'初期ソート'}
            selectItems={[new SelectItem('選択1', 'ラベル名', ['選択1', '選択2'])]}
            linkValues={playerIdMap}
            path={'/path/'}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
