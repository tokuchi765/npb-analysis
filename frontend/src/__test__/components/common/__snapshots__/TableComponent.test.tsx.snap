// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponentテスト スナップショット作成 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
    id="tableTitle"
  >
    <div
      className="MuiGrid-root makeStyles-grid-6 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root makeStyles-paper-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          選択1_
          タイトル
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink={true}
            id="demo-simple-select-label"
          >
            ラベル名
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="demo-simple-select"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              選択1
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="選択1"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
          scope="col"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            チーム名
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            打率
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        role={null}
        tabIndex={-1}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
          id="enhanced-table-checkbox-0"
          role="cell"
          scope="row"
        >
          <div>
            メイン
          </div>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0.3
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableLinkComponentテスト スナップショット作成 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
    id="tableTitle"
  >
    <div
      className="MuiGrid-root makeStyles-grid-6 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiPaper-root makeStyles-paper-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          選択1_
          タイトル
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink={true}
            id="demo-simple-select-label"
          >
            ラベル名
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="demo-simple-select"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              選択1
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="選択1"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
          scope="col"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            チーム名
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            打率
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        role={null}
        tabIndex={-1}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
          id="enhanced-table-checkbox-0"
          role="cell"
          scope="row"
        >
          <a
            href="/path/undefined"
            onClick={[Function]}
          >
            メイン
          </a>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0.3
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
